include/seqcmd.h

```diff
@@ -0,0 +1,652 @@
+    /* 0x2 */ SEQ_SUB_CMD_TEMPO_SLOW_DOWN,
+    /* 0x3 */ SEQ_SUB_CMD_TEMPO_SCALE,
+    /* 0x4 */ SEQ_SUB_CMD_TEMPO_RESET
+} TempoCmdType;
+
+// Subset of `SEQ_CMD_SETUP_CMD`
+typedef enum {
+    /* 0x0 */ SEQ_SUB_CMD_SETUP_RESTORE_VOLUME,
+    /* 0x1 */ SEQ_SUB_CMD_SETUP_SEQ_UNQUEUE,
+    /* 0x2 */ SEQ_SUB_CMD_SETUP_RESTART_SEQ,
+    /* 0x3 */ SEQ_SUB_CMD_SETUP_TEMPO_SCALE,
+    /* 0x4 */ SEQ_SUB_CMD_SETUP_TEMPO_RESET,
+    /* 0x5 */ SEQ_SUB_CMD_SETUP_PLAY_SEQ,
+    /* 0x6 */ SEQ_SUB_CMD_SETUP_SET_FADE_TIMER,
+    /* 0x7 */ SEQ_SUB_CMD_SETUP_RESTORE_VOLUME_IF_QUEUED,
+    /* 0x8 */ SEQ_SUB_CMD_SETUP_RESTORE_VOLUME_WITH_SCALE,
+    /* 0x9 */ SEQ_SUB_CMD_SETUP_DISABLE_CHANNELS,
+    /* 0xA */ SEQ_SUB_CMD_SETUP_SET_PLAYER_FREQ,
+    /* 0xE */ SEQ_SUB_CMD_SETUP_POP_CACHE = 0xE,
+    /* 0xF */ SEQ_SUB_CMD_SETUP_RESET_SETUP_CMDS
+} SetupCmdType;
+
+// Subset of `SEQ_CMD_GLOBAL_CMD`
+typedef enum {
+    /* 0x0 */ SEQ_SUB_CMD_GLOBAL_SET_SOUND_MODE,
+    /* 0x1 */ SEQ_SUB_CMD_GLOBAL_DISABLE_NEW_SEQUENCES
+} SubCmdType;
+
+// ==== Audio Sequence Primary Commands ====
+
+/**
+ * ARGS
+ *   playerIndex (p), fadeInTimer (t), seqArgs (a), seqId (s)
+ *
+ * FORMAT
+ *   Capital U is unused
+ *   0pttaass
+ *
+ * DESCRIPTION
+ *   Request a sequence (`seqId`) to be played on the specified player (`playerIndex`).
+ *   The sequence will gradually fade in over the course of (`fadeInTimer`).
+ *   How (`fadeInTimer`) is interpreted depends on (`seqArgs`)
+ *
+ *   `seqArgs = 8` and `seqArgs = 9` (commonly used arguments) appear to have no functionality,
+ *   although unused commands may suggest  these were intended to be a an priority for an unimplemented priority system.
+ *
+ *   `((seqArgs & 0x7F) != 0x7F)` will interpret fadeInTimer as number of frames at 30 fps
+ *   `((seqArgs & 0x7F) == 0x7F)` will interpret fadeInTimer as skip ticks
+ *   `(seqArgs & 0x80)` i.e. `(seqArgs >= 0x80)` has an incomplete implementation
+ *   and is not functional in Ocarina of Time. No sequence will play as a result.
+ *   The code and purpose for `seqArgs >= 0x80` was completed and is functional in Majora's Mask.
+ */
+#define SEQCMD_PLAY_SEQUENCE(playerIndex, fadeInTimer, seqArgs, seqId)                               \
+    Audio_QueueSeqCmd((SEQ_CMD_PLAY << 28) | ((u8)(playerIndex) << 24) | ((u8)(fadeInTimer) << 16) | \
+                      ((u8)(seqArgs) << 8) | (u16)(seqId))
+
+/**
+ * ARGS
+ *   playerIndex (p), fadeOutTimer (t)
+ *
+ * FORMAT
+ *   Capital U is unused
+ *   1pttUUFF
+ *
+ * DESCRIPTION
+ *   Request the active sequence to be stopped on the specified player (`playerIndex`).
+ *   The sequence will gradually fade out over the course of (`fadeOutTimer`) frames at a rate of 30 fps.
+ *   Note: The 0xFF in the command is not read from at all, but is common in all StopSequence Commands
+ */
+#define SEQCMD_STOP_SEQUENCE(playerIndex, fadeOutTimer) \
+    Audio_QueueSeqCmd((SEQ_CMD_STOP << 28) | 0xFF | ((u8)(playerIndex) << 24) | ((u8)(fadeOutTimer) << 16))
```

Comment:
`fadeOutDuration` may be a better way to describe `fadeOutTimer`

at least in the contexts it gets used (mainly looking at z_demo), `endFrame - startFrame` gives you the duration, how long the fade will take. It itself is not a timer

Reply:
looks like this probably applies to a few other commands here

Reply:
audio at the moment is a bit inconsistent with its usage of `fadeOut` and `duration` but both are commonly found. I've used `duration` in most commands where I could have chosen either, but this `fadeTimer` goes into a variable named `fadeTimer` (which I'm happy to change into duration, but that would extend outside of this file and I'd rather leave that to a separate PR).

For now though, I could just change this to `duration`? (Or would you still prefer the full `fadeOutDuration`)?

Reply:
since the command is called stop sequence, it wouldnt be clear what `duration` on its own means.
But yeah, we can figure it out later when looking to make duration vs timer consistent in audio code.

think in general that inconsistency isnt too much of an issue, the variable it gets set to is the timer itself, so it makes sense to be called that. But the value that will get assigned to the timer is a duration

but yea, doesnt have to be figured out in this PR